### DISCUSSION TITLE
Uploading the import csv for products.

### DISCUSSION THREAD

**Discussion by jake1808** [medusajs/medusa#4811] [Q&A] [answered]
Hello, I'm trying to use the csv template fund in the products module to upload products. I have configured my backend to use Spaces. When uploading the csv I encounter the following error:

```
error:   An error occurred while processing batch.confirmed: Error: Error while processing row with
      [(product id: undefined),
      (product handle: 3M Spray Gun Gravity Hvhp Acrf 1.4Mm Sgh827),
      (variant id: undefined),
      (variant sku: 27459)]:
      duplicate key value violates unique constraint "PK_1cf5c9537e7198df494b71b993f"
```
I have checked the CSV for duplicates and they aren't any. Maybe I\m filling in the CSV wrong if any one could help I would highly appreciate it.
P.S I have attached a copy of the CSV file. 
[product-import-sales-channels_edit3.csv](https://github.com/medusajs/medusa/files/12386204/product-import-sales-channels_edit3.csv)

**Comment by fPolic** [comment, marked as answer]
hey @jake1808, the issue is that in the "Product Tags" column you have redundant values. For example in the row that throws the error there are duplicates for "3M" and "Tools".

**Reply by jake1808** [reply]
Sorry one more thing about the import. my prices are popping as NaN on my front-end NextJs template. I can't find the pricing in the DB.

**Reply by jake1808** [reply]
```
Unexpected Application Error!
Cannot read properties of undefined (reading 'decimal_digits')
TypeError: Cannot read properties of undefined (reading 'decimal_digits')
    at MedusaPriceInput (webpack-internal:///91225:68:35)
    at renderWithHooks (webpack-internal:///91344:16305:18)
    at mountIndeterminateComponent (webpack-internal:///91344:20069:13)
    at beginWork (webpack-internal:///91344:21582:16)
    at beginWork$1 (webpack-internal:///91344:27421:14)
    at performUnitOfWork (webpack-internal:///91344:26552:12)
    at workLoopSync (webpack-internal:///91344:26461:5)
    at renderRootSync (webpack-internal:///91344:26429:7)
    at recoverFromConcurrentError (webpack-internal:///91344:25845:20)
    at performSyncWorkOnRoot (webpack-internal:///91344:26091:20)
💿 Hey developer 👋

You can provide a way better UX than this when your app throws errors by providing your own errorElement props on <Route>
```
This is the error I faced when trying to add in the prices manually

**Reply by fPolic** [reply]
Have you configured currencies in the admin?

**Comment by jake1808** [comment]
Thanks @fPolic  your suggestion worked. I would also like to find out the import limit if you would happen to know?

**Reply by fPolic** [reply]
There is no limit on the number of imported products/variants, you are only limited by file/database storage.
